### DISCUSSION TITLE
Python comments sync for readme

### DISCUSSION THREAD

**Discussion by quietCoder500** [community/community#62183] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Hello! I was writing a Django project and commenting something I needed to remember for later when I thought, "What if Github could scan a .py file and take certain comments and say, auto create a todo list?"
Maybe like this: 
`# TODO: require Django Form C to choose A or B`

then:

README.md
--------------
Todos:
-----1: require Django Form C to choose A or B

or something similar.
Is this possible or could it be something for the future?
      Thanks,
       [quietCoder500](https://github.com/quietCoder500)

**Comment by ikyawthetpaing** [comment]
Hello! Your idea of having GitHub scan a .py file to auto-create a TODO list based on certain comments is interesting and could be quite useful for developers. While GitHub itself doesn't currently have a built-in feature to automatically generate a TODO list from comments in Python files, there are some existing tools and practices that can help achieve a similar outcome.

-     Task List in Markdown: GitHub already supports task lists in Markdown. You can manually create a TODO list in your README.md file and keep it in sync with your code comments. For example:

```
Todos:
-----1. [ ] require Django Form C to choose A or B
```

You can update the checkbox manually in the README.md when you complete a task.

-     Third-Party Tools: There are external tools and integrations that can help you automate the process to some extent. For instance, you can use task management tools like Trello, Asana, or Jira, which can integrate with GitHub. Some of these tools have integrations that can parse your codebase for TODO comments and automatically create tasks for you. It might not be a direct GitHub feature, but it can accomplish what you're looking for.

-     Custom Scripts: If you have more specific requirements and are comfortable with scripting, you could write a custom script to parse your Python files for TODO comments and generate a markdown file for your GitHub repository automatically. Python itself has libraries like re (regular expressions) that can help you find and extract TODO comments from the code files.

-     Feature Requests: While this specific feature isn't available in GitHub right now, the platform is continually evolving. You can consider submitting a feature request to GitHub or raising the idea in their community forums. If it gains enough traction or aligns with their roadmap, there's a chance they might implement it in the future.

Remember, even though the exact functionality you mentioned might not be available natively in GitHub at the moment, there are still several ways you can achieve a similar result using existing tools and a bit of customization.

**Comment by look** [comment, marked as answer]
You could try using [GitHub Actions](https://docs.github.com/en/actions) to automate this.
